### ISSUE TITLE
JSONDecodeError

### ISSUE THREAD

**Issue by caullla** [django-salesforce/django-salesforce#181]
Hi,
Not sure whats goes wrong, but we get this error multiple times.
It crashed when we execute `Select` query on `Account` table.

```
JSONDecodeError: Expecting value: line 3 column 1 (char 2)
  File "django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "braces/views/_access.py", line 102, in dispatch
    request, *args, **kwargs)
  File "braces/views/_access.py", line 201, in dispatch
    request, *args, **kwargs)
  File "formtools/wizard/views.py", line 237, in dispatch
    response = super(WizardView, self).dispatch(request, *args, **kwargs)
  File "django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "mypp/views.py", line 2545, in get
    return super(CustomerCreateWizard, self).get(*args, **kwargs)
  File "mypp/common/util/views.py", line 26, in get
    if self.get_cleaned_data_for_step(step) is None:
  File "formtools/wizard/views.py", line 496, in get_cleaned_data_for_step
    files=self.storage.get_step_files(step))
  File "mypp/views.py", line 2666, in get_form
    form = super(CustomerCreateWizard, self).get_form(step, data, files)
  File "mypp/common/util/views.py", line 79, in get_form
    form = super(GenericWizardView, self).get_form(step, data, files)
  File "formtools/wizard/views.py", line 415, in get_form
    'initial': self.get_form_initial(step),
  File "mypp/views.py", line 2692, in get_form_initial
    initial.update(self._get_initial__location(step))
  File "mypp/views.py", line 2702, in _get_initial__location
    stroeer_contract_number = self._get_attr_value_for_slug(str(step).split('_')[1], 'stroeer_contract_number')
  File "mypp/views.py", line 2738, in _get_attr_value_for_slug
    data = self.get_cleaned_data_for_step('locations') or []
  File "formtools/wizard/views.py", line 497, in get_cleaned_data_for_step
    if form_obj.is_valid():
  File "django/forms/formsets.py", line 316, in is_valid
    self.errors
  File "django/forms/formsets.py", line 290, in errors
    self.full_clean()
  File "django/forms/formsets.py", line 339, in full_clean
    self._errors.append(form.errors)
  File "django/forms/forms.py", line 153, in errors
    self.full_clean()
  File "django/forms/forms.py", line 362, in full_clean
    self._clean_fields()
  File "django/forms/forms.py", line 380, in _clean_fields
    value = field.clean(value)
  File "django/forms/fields.py", line 168, in clean
    self.run_validators(value)
  File "django/forms/fields.py", line 151, in run_validators
    v(value)
  File "my_app_inner/common/util/validators.py", line 59, in __call__
    models.Account.objects.get(Id=value)
  File "django/db/models/manager.py", line 122, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "django/db/models/query.py", line 381, in get
    num = len(clone)
  File "django/db/models/query.py", line 240, in __len__
    self._fetch_all()
  File "django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "salesforce/backend/query.py", line 269, in iterator
    cursor.execute(sql, params)
  File "salesforce/backend/query.py", line 424, in execute
    response = self.execute_select(q, args)
  File "salesforce/backend/query.py", line 474, in execute_select
    return handle_api_exceptions(url, self.session.get, _cursor=self)
  File "salesforce/backend/query.py", line 132, in handle_api_exceptions
    data = response.json()[0]
  File "requests/models.py", line 866, in json
    return complexjson.loads(self.text, **kwargs)
  File "simplejson/__init__.py", line 516, in loads
    return _default_decoder.decode(s)
  File "simplejson/decoder.py", line 374, in decode
    obj, end = self.raw_decode(s)
  File "simplejson/decoder.py", line 404, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
```

**Comment by caullla** [comment]
I just noticed that we use not the latest version, we use 0.6.3.
I'll check it update will fix the error.

**Comment by hynekcer** [comment]
It is fixed in the first commit after 0.6.9: fb42ab8469. You can upgrade to this commit or to the current master branch. Our problem was that most of error messages from Salesforce especially SOQL errors are reported nice in JSON format, but some ugly errors ere reported by plain text and http status code. If I remember, that are e.g. too long request, too many requests, authentication errors. You should get helpful error messages after applying the fix.
